concepts/random/about.md

```diff
@@ -0,0 +1,156 @@
+
+The first way to generate a value is to create a `Random.Seed` via `initialSeed : Int -> Seed`.
+A `Seed` is an opaque type that contains an integer and knows how to transform that integer in a way that will appear random (called pseudorandom).
+Once you have a `Seed`, you can use it to generate a value with a `Generator a`.
+A `Generator a` is an opaque type that knows how to use the integer inside of a seed to create a pseudorandom value of type `a` as well as a new seed with an updated integer.
+
+To generate a value, we can use `step : Generator a -> Seed -> ( a, Seed )`, which returns a value and a new seed that we can use to generate further values.
+
+Let's use these functions to extract `n` random values out of a generator:
+
+```elm
+generate : Int -> Generator a -> List a
+generate n generator = generateValuesFromSeed n generator (Random.initialSeed 42)
+
+generateValuesFromSeed : Int -> Generator a -> Seed -> List a
+generateValuesFromSeed n generator seed =
+    if n <= 0 then
+        []
+    else
+        let ( value, nextSeed ) = Random.step generator seed
+        in value :: generateValuesFromSeed (n - 1) generator nextSeed
+```
+
+Note that all of these functions are pure, so calling them twice with the same arguments will produce the same values.
+
+```elm
+generate 10 (Random.int 1 6)
+    --> [4, 5, 3, 3, 5, 1, 2, 4, 6, 6]
+
+generate 10 (Random.int 1 6)
+    --> [4, 5, 3, 3, 5, 1, 2, 4, 6, 6]
+```
+
+The second way to generate a value is by using `Random.generate : (a -> msg) -> Generator a -> Cmd msg`, but that can only be done inside an Elm application.
+In that case, the Elm runtime may use `step` as well as outside, non-pure resources to generate seeds, and calling the same function twice will give different results.
+
+For now on, we will focus on generators.
+
+The `Random` module provides two basic generators for creating integers and floats within a specific range:
+
+```elm
+generate 5 (Random.int -5 5)
+    --> [0, 3, -5, 5, 0]
+
+generate 3 (Random.float 0 5)
+    --> [1.61803, 3.14159, 2.71828]
+```
+
+Those values can be combined into pairs or lists of values:
+
+```elm
+generate 2 (Random.list 3 (Random.int 0 3))
+    --> [[0, 3, 3], [1, 3, 2]]
+
+generate 2 (Random.pair (Random.int 0 3) (Random.float 10 10.3))
+    --> [(0, 10.23412), (2, 10.17094)]
+```
+
+The (`elm-community/random-extra`)[random-extra] package provides a lot more generators for various data structures: strings, dates, dictionaries, arrays, sets, etc.
+
+We can create generators that will only return a single value using `Random.constant`:
+
+```elm
+generate 4 (Random.constant "hello")
+    --> ["hello", "hello", "hello", "hello"]
+```
+
+We can randomly pick from given elements with equal probability using `Random.uniform`:
+
+```elm
+generate 5 (Random.uniform Red [Green, Blue])
+    --> [Red, Blue, Blue, Green, Red]
+```
+
+`Random.uniform` takes two arguments to guarantee that there is at least one value to pick from.
```

Comment:
```suggestion
`Random.uniform` takes two arguments (`Red, [Green, Blue]`) to guarantee that there is at least one value to pick from.
Otherwise if it took one list argument (`[Red, Green, Blue`]) then the list could be empty.
```